File: Components/WinForms/CS/TreeListTutorials/NodesNavigation/NodesNavigation.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace DevExpress.XtraTreeList.Demos.Tutorials {
    /// <summary>
    /// Summary description for NodesNavigation.
    /// </summary>
    public partial class NodesNavigation : TutorialControl {
        public NodesNavigation() {
            //
            // Required for Windows Form Designer support
            //
            InitializeComponent();
            //
            // TODO: Add any constructor code after InitializeComponent call
            //
        }

        //<button1>
        private void button1_Click(object sender, System.EventArgs e) {
            treeList1.MoveFirst();
        }
        //</button1>

        //<button2>
        private void button2_Click(object sender, System.EventArgs e) {
            treeList1.MovePrev();
        }
        //</button2>

        //<button3>
        private void button3_Click(object sender, System.EventArgs e) {
            treeList1.MovePrevVisible();
        }
        //</button3>

        //<button4>
        private void button4_Click(object sender, System.EventArgs e) {
            treeList1.MoveNext();
        }
        //</button4>

        //<button5>
        private void button5_Click(object sender, System.EventArgs e) {
            treeList1.MoveNextVisible();
        }
        //</button5>

        //<button6>
        private void button6_Click(object sender, System.EventArgs e) {
            treeList1.MoveLast();
        }
        //</button6>

        //<button7>
        private void button7_Click(object sender, System.EventArgs e) {
            treeList1.MoveLastVisible();
        }
        //</button7>
    }
}

File: Components/WinForms/VB/TreeListTutorials/NodesNavigation/NodesNavigation.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data

Namespace DevExpress.XtraTreeList.Demos.Tutorials
	''' <summary>
	''' Summary description for NodesNavigation.
	''' </summary>
	Public Partial Class NodesNavigation
		Inherits TutorialControl
		Public Sub New()
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()
			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		'<button1>
		Private Sub button1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button1.Click
			treeList1.MoveFirst()
		End Sub
		'</button1>

		'<button2>
		Private Sub button2_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button2.Click
			treeList1.MovePrev()
		End Sub
		'</button2>

		'<button3>
		Private Sub button3_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button3.Click
			treeList1.MovePrevVisible()
		End Sub
		'</button3>

		'<button4>
		Private Sub button4_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button4.Click
			treeList1.MoveNext()
		End Sub
		'</button4>

		'<button5>
		Private Sub button5_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button5.Click
			treeList1.MoveNextVisible()
		End Sub
		'</button5>

		'<button6>
		Private Sub button6_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button6.Click
			treeList1.MoveLast()
		End Sub
		'</button6>

		'<button7>
		Private Sub button7_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles button7.Click
			treeList1.MoveLastVisible()
		End Sub
		'</button7>
	End Class
End Namespace
